crepecake/src/test/java/com/google/cloud/tools/crepecake/image/DescriptorDigestTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+/** Tests for {@link DescriptorDigest}. */
+public class DescriptorDigestTest {
+
+  @Test
+  public void testCreateFromHash_pass() throws DigestException {
+    String goodHash = createGoodHash('a');
+
+    DescriptorDigest descriptorDigest = DescriptorDigest.fromHash(goodHash);
+
+    Assert.assertEquals(goodHash, descriptorDigest.getHash());
+    Assert.assertEquals("sha256:" + goodHash, descriptorDigest.toString());
+  }
+
+  @Test
+  public void testCreateFromHash_fail() {
+    String badHash = "not a valid hash";
+
+    try {
+      DescriptorDigest.fromHash(badHash);
+      Assert.fail("Invalid hash should have caused digest creation failure.");
+    } catch (DigestException ex) {
+      // pass
+    }
+  }
+
+  @Test
+  public void testCreateFromDigest_pass() throws DigestException {
+    String goodHash = createGoodHash('a');
+    String goodDigest = "sha256:" + createGoodHash('a');
+
+    DescriptorDigest descriptorDigest = DescriptorDigest.fromDigest(goodDigest);
+
+    Assert.assertEquals(goodHash, descriptorDigest.getHash());
+    Assert.assertEquals(goodDigest, descriptorDigest.toString());
+  }
+
```

Comment:
I tested this myself to make sure, but it might help to have a test that ensure a valid character matching but invalid length still fails.

Reply:
Added the test.